Generate and move to using DT_ prefixed FLASH defines

galak opened the pull request.

galak committed:
scripts/dts/extract_dts_includes: Add DT_ prefix to FLASH defines

We generate the following FLASH prefixed defines:
	FLASH_LABEL
	FLASH_ERASE_BLOCK_SIZE
	FLASH_WRITE_BLOCK_SIZE

Now we generate DT_ prefixed versions:
	DT_FLASH_LABEL
	DT_FLASH_ERASE_BLOCK_SIZE
	DT_FLASH_WRITE_BLOCK_SIZE

And will deprecate the non DT_ prefixed versions in the future.  We keep
them around as aliases for the time being.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
scripts/dts/extract_dts_includes: Add DT_ prefix to FLASH_AREA defines

We generate the following FLASH prefixed defines:

	FLASH_AREA_MCUBOOT_LABEL
	FLASH_AREA_MCUBOOT_OFFSET
	FLASH_AREA_MCUBOOT_OFFSET_0
	FLASH_AREA_MCUBOOT_READ_ONLY
	FLASH_AREA_MCUBOOT_SIZE
	FLASH_AREA_MCUBOOT_SIZE_0

Now we generate DT_ prefixed versions:

	DT_FLASH_AREA_MCUBOOT_LABEL
	DT_FLASH_AREA_MCUBOOT_OFFSET
	DT_FLASH_AREA_MCUBOOT_OFFSET_0
	DT_FLASH_AREA_MCUBOOT_READ_ONLY
	DT_FLASH_AREA_MCUBOOT_SIZE
	DT_FLASH_AREA_MCUBOOT_SIZE_0

And will deprecate the non DT_ prefixed versions in the future.  We keep
them around as aliases for the time being.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
boards: reel_board: Convert FLASH_AREA to DT_FLASH_AREA

Use DT_FLASH_AREA prefixed defined instead of FLASH_AREA as the non-DT
version is deprecated.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
samples: nvs: Convert FLASH_AREA to DT_FLASH_AREA

Use DT_FLASH_AREA prefixed defined instead of FLASH_AREA as the non-DT
version is deprecated.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
fs: nffs: Convert FLASH_AREA to DT_FLASH_AREA

Use DT_FLASH_AREA prefixed defined instead of FLASH_AREA as the non-DT
version is deprecated.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
west: sign: Convert FLASH_AREA to DT_FLASH_AREA

Use DT_FLASH_AREA prefixed defined instead of FLASH_AREA as the non-DT
version is deprecated.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
dfu: boot: mcuboot: Convert FLASH_AREA to DT_FLASH_AREA

Use DT_FLASH_AREA prefixed defined instead of FLASH_AREA as the non-DT
version is deprecated.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
tests: fs: multi-fs: Convert FLASH_AREA to DT_FLASH_AREA

Use DT_FLASH_AREA prefixed defined instead of FLASH_AREA as the non-DT
version is deprecated.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
flash: flash_stm32: Convert FLASH_WRITE_BLOCK_SIZE to DT_

Convert generic FLASH_WRITE_BLOCK_SIZE to driver specific generated
define DT_SOC_NV_FLASH_0_WRITE_BLOCK_SIZE.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
flash: soc_flash_mcux: Convert FLASH_ERASE_BLOCK_SIZE to DT_

Convert generic FLASH_ERASE_BLOCK_SIZE to driver specific generated
define DT_SOC_NV_FLASH_0_ERASE_BLOCK_SIZE.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
flash: flash_sam: Convert FLASH_{WRITE,ERASE}_BLOCK_SIZE to DT_

Convert generic FLASH_{WRITE,ERASE}_BLOCK_SIZE to driver specific
generated define DT_SOC_NV_FLASH_0_{WRITE,ERASE}_BLOCK_SIZE.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
flash: flash_gecko: Convert FLASH_WRITE_BLOCK_SIZE to DT_

Convert generic FLASH_WRITE_BLOCK_SIZE to driver specific generated
define DT_SOC_NV_FLASH_0_WRITE_BLOCK_SIZE.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
flash: spi_nor: Convert FLASH_{WRITE,ERASE}_BLOCK_SIZE to DT_

Convert generic FLASH_{WRITE,ERASE}_BLOCK_SIZE to driver specific
generated define DT_JEDEC_SPI_NOR_0_{WRITE,ERASE}_BLOCK_SIZE.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
scripts/dts/extract: Remove generation for FLASH_LABEL

FLASH_LABEL define isn't used anywhere so remove it.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
samples: subsys: nvs: Convert FLASH_ERASE_BLOCK_SIZE to DT_

Use DT_FLASH_ERASE_BLOCK_SIZE prefixed defined instead of
FLASH_ERASE_BLOCK_SIZE as the non-DT version is deprecated.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
openthread: Convert FLASH_ERASE_BLOCK_SIZE to DT_

Use DT_FLASH_ERASE_BLOCK_SIZE prefixed defined instead of
FLASH_ERASE_BLOCK_SIZE as the non-DT version is deprecated.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
west: sign: Convert FLASH_WRITE_BLOCK_SIZE to DT_

Use DT_FLASH_WRITE_BLOCK_SIZE prefixed defined instead of
FLASH_WRITE_BLOCK_SIZE as the non-DT version is deprecated.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
dfu: boot: mcuboot: Convert FLASH_WRITE_BLOCK_SIZE to DT_

Use DT_FLASH_WRITE_BLOCK_SIZE prefixed defined instead of
FLASH_WRITE_BLOCK_SIZE as the non-DT version is deprecated.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
dfu: img_util: Convert FLASH_WRITE_BLOCK_SIZE to DT_

Use DT_FLASH_WRITE_BLOCK_SIZE prefixed defined instead of
FLASH_WRITE_BLOCK_SIZE as the non-DT version is deprecated.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

mbolivar commented on scripts/dts/extract/flash.py:
```diff
@@ -85,6 +85,9 @@ def _extract_partition(self, node_address):
         insert_defs("DT_FLASH_AREA", prop_def, prop_alias)
 
+    def _create_legacy_label(self, prop_alias, label):
+        prop_alias[label[3:]] = label
```
Maybe `label.lstrip('DT_')` would be clearer?

galak replied:
Thanks, made the change.

mbolivar reviewed (approved):
Fine by me as long as the old defines are still around for v1.14

galak commented:
> Fine by me as long as the old defines are still around for v1.14

They will be, I want to mark them with `__DEPRECATED_MACRO`

mbolivar reviewed (changes requested):
@galak doing some testing and `west sign` was broken by 69c15f0ed1 (just 10 patches after it was merged...) and is still broken with these patches. I'm digging but want to -1 this for now until the problem is found

galak commented:
> @galak doing some testing and `west sign` was broken by [69c15f0](https://github.com/zephyrproject-rtos/zephyr/commit/69c15f0ed17e1c74e1288b5bf7f99aeef829afeb) (just 10 patches after it was merged...) and is still broken with these patches. I'm digging but want to -1 this for now until the problem is found

Ok, at least 10 patches is a small amount to bisect :)

mbolivar commented:
@galak this is not the root cause of west sign not working, but this PR has an issue. Take a look at the generated_dts_board.conf: 

https://hastebin.com/amedewumuv.ini

A lot of the 'legacy' defines are the string "None" instead of the number they should be.

mbolivar commented:
> Ok, at least 10 patches is a small amount to bisect :)

That *is* the first bad patch; it was broken almost immediately haha

galak committed:
tests: fs: nffs_fs_api: Convert FLASH_AREA to DT_FLASH_AREA

Use DT_FLASH_AREA prefixed defined instead of FLASH_AREA as the non-DT
version is deprecated.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
scripts/dts/extract_dts_includes: Hand alias to alias for write_conf

If we have an alias pointing to another alias we would get None in the
resulting conf file.  However that alias might exist, so try looking it
up again in the aliases db.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
scripts/dts/extract: Fix producing bogus alias if reg has no size

Its possible that a reg property is address only.  If that's the case we
should produce any alias information for the size.  We should check
nr_size_cells to decide to handle the alias case or not.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

mbolivar reviewed (approved):
Retested the latest (2d2dfdb51401ddbc4d6348dcf3251129fe77a162) with https://github.com/JuulLabs-OSS/mcuboot/pull/405 applied. Looks good. Visually inspected the generated defines, including the legacy defines, and they are consistent with one another. Tested west sign still works.

mbolivar commented:
recheck
